contracts/peripheral/GaugeVoteRewards.sol

```diff
@@ -0,0 +1,318 @@
+    }
+
+    function setIsNoDepositGauge(address _gauge, bool _isNoDeposit) external onlyOwner {
+        isNoDepositGauge[_gauge] = _isNoDeposit;
+        emit SetIsNoDepositGauge(_gauge, _isNoDeposit);
+    }
+
+    function setDstChainId(address[] memory _gauges, uint16[] memory _dstChainIds) external onlyOwner {
+        uint256 dstChainIdsLen = _dstChainIds.length;
+        require(dstChainIdsLen == _gauges.length, "!length");
+        for (uint256 i = 0; i < dstChainIdsLen; i++) {
+            getDstChainId[_gauges[i]] = _dstChainIds[i];
+        }
+    }
+
+    function setChildGaugeVoteRewards(uint16[] memory _dstChainIds, address[] memory _voteRewards) external onlyOwner {
+        uint256 dstChainIdsLen = _dstChainIds.length;
+        require(dstChainIdsLen == _voteRewards.length, "!length");
+        for (uint256 i = 0; i < dstChainIdsLen; i++) {
+            getChildGaugeVoteRewards[_dstChainIds[i]] = _voteRewards[i];
+        }
+    }
+
+    function setPoolIds(uint256[] memory _poolIds) external {
+        uint256 poolIdsLen = _poolIds.length;
+        for (uint256 i = 0; i < poolIdsLen; i++) {
+            uint256 pid = _poolIds[i];
+            IBooster.PoolInfo memory poolInfo = booster.poolInfo(pid);
+            getPoolId[poolInfo.gauge] = pid;
+        }
+    }
+
+    /* -------------------------------------------------------------------
+       View 
+    ------------------------------------------------------------------- */
+
+    function getCurrentEpoch() external view returns (uint256) {
+        return _getCurrentEpoch();
+    }
+
+    /* -------------------------------------------------------------------
+       Core 
+    ------------------------------------------------------------------- */
+
+    /**
+     * @notice  Wraps the booster.voteGaugeWeight call to track weights for each epoch
+     *          So AURA rewards can be distributed pro rata to those pool stashes
+     * @param _gauge    Array of the gauges
+     * @param _weight   Array of the weights
+     * @return bool for success
+     */
+    function voteGaugeWeight(address[] calldata _gauge, uint256[] calldata _weight) external onlyOwner returns (bool) {
+        uint256 totalWeight = 0;
+        uint256 totalDepositWeight = 0;
+        uint256 epoch = _getCurrentEpoch();
+
+        // Loop through each gauge and store it's weight for this epoch
+        for (uint256 i = 0; i < _gauge.length; i++) {
+            address gauge = _gauge[i];
+            uint256 weight = _weight[i];
+
+            totalWeight = totalWeight.add(weight);
+
+            // Some gauges like veBAL have no deposits so we just skip
+            // those special cases
+            if (isNoDepositGauge[gauge]) continue;
+
+            getWeightByEpoch[epoch][gauge] = weight;
+            totalDepositWeight = totalDepositWeight.add(weight);
+        }
+
+        // Update the total weight for this epoch
+        getTotalWeight[epoch] = getTotalWeight[epoch].add(totalWeight);
```

Comment:
Should not be like this ? 
```
getTotalWeight[epoch] =  totalWeight;
```

If it is called more than once then this value will continue increasing